services/eventservice/src/tests.rs

```diff
@@ -0,0 +1,508 @@
+        .expect("Failed to create category");
+    assert_eq!(
+        manager.get_categories().unwrap(),
+        vec![cat1.clone(), cat2.clone(), cat3.clone()]
+    );
+    manager
+        .delete_category(&cat1)
+        .expect("Failed to delete category");
+    assert!(manager.delete_category(&cat4).is_err());
+    assert_eq!(manager.get_categories().unwrap(), vec![cat2, cat3]);
+    manager._reset_all().expect("Failed to delete db");
+}
+
+#[test]
+fn merge_categories_test() {
+    let manager = data_manager::EventManager::new(&"./test/mergetestdb".to_string());
+    let cat1 = Category {
+        name: "first".to_string(),
+    };
+    let cat2 = Category {
+        name: "second".to_string(),
+    };
+    let event1 = types::Event {
+        id: 1,
+        title: "Event1".to_string(),
+        cover: "N/A".to_string(),
+        description: "Empty".to_string(),
+        organizer: 1,
+        date_created: chrono::Utc::now().timestamp(),
+        date_planning: (chrono::Utc::now() + chrono::Duration::days(3)).timestamp(),
+        category: cat1.clone(),
+    };
+    let event2 = types::Event {
+        id: 2,
+        title: "Event2".to_string(),
+        cover: "N/A".to_string(),
+        description: "Empty".to_string(),
+        organizer: 1,
+        date_created: chrono::Utc::now().timestamp(),
+        date_planning: (chrono::Utc::now() + chrono::Duration::days(2)).timestamp(),
+        category: cat2.clone(),
+    };
+    let event3 = types::Event {
+        id: 3,
+        title: "Event3".to_string(),
+        cover: "N/A".to_string(),
+        description: "Empty".to_string(),
+        organizer: 2,
+        date_created: chrono::Utc::now().timestamp() + 10,
+        date_planning: (chrono::Utc::now() + chrono::Duration::days(1)).timestamp(),
+        category: cat1.clone(),
+    };
+    manager
+        .create_category(&cat1)
+        .expect("Failed to create category!");
+    manager
+        .create_category(&cat2)
+        .expect("Failed to create category!");
+    manager
+        .create_event(&event1)
+        .expect("Failed to create event");
+    manager
+        .create_event(&event2)
+        .expect("Failed to create event");
+    manager
+        .create_event(&event3)
+        .expect("Failed to create event");
+    assert_eq!(
+        manager
+            .get_events(Some(&"title".to_string()), None, Some(&cat2.name))
+            .unwrap(),
+        vec![event2.clone()]
+    );
+    assert_eq!(
+        manager
+            .get_events(Some(&"title".to_string()), None, Some(&cat1.name))
+            .unwrap(),
+        vec![event1.clone(), event3.clone()]
+    );
+    manager
+        .merge_categories(&cat1, &cat2)
+        .expect("Failed to merge");
+    assert_eq!(
+        manager
+            .get_events(Some(&"title".to_string()), None, Some(&cat1.name))
+            .unwrap(),
+        vec![event1, event2, event3]
+    );
+    manager._reset_all().expect("Failed to delete db");
+}
```

Comment:
There are too much in the one file. Maybe if you split it up a little bit it will be easier to read.

Reply:
Where is the main entry point for your tests?
Do you have any script to run it?

Reply:
I think 500 lines is not too much for one file
Entry point is in main.rs file, you can run it with 
`cargo test`